### PR TITLE
[SPARK-48923][SQL][TESTS] Fix the incorrect logic of `CollationFactorySuite`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to fix the incorrect logic of `CollationFactorySuite`.

### Why are the changes needed?
Only fix `CollationFactorySuite`.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Update existed UT.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by dongjoon-hyun — DISMISSED** [review]
Please use a proper JIRA ID for code change. Especially, this is a kind of `Fix`.

File: common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala [hunk 1]
```diff
@@ -154,8 +151,8 @@ class CollationFactorySuite extends AnyFunSuite with Matchers { // scalastyle:ig
         testCase.expectedResult)
 
       val hash1 = collation.hashFunction.applyAsLong(toUTF8(testCase.s1))
-      val hash2 = collation.hashFunction.applyAsLong(toUTF8(testCase.s1))
-      assert(hash1 == hash2)
+      val hash2 = collation.hashFunction.applyAsLong(toUTF8(testCase.s2))
+      assert((hash1 == hash2) == testCase.expectedResult)
```
Comment (panbingkun) [inline common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala:155]: Fix it

**Comment by panbingkun** [comment]
> Please use a proper JIRA ID for code change. Especially, this is a kind of `Fix`.

Okay, let me file it.

File: common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala [hunk 2]
```diff
@@ -154,8 +151,8 @@ class CollationFactorySuite extends AnyFunSuite with Matchers { // scalastyle:ig
         testCase.expectedResult)
 
       val hash1 = collation.hashFunction.applyAsLong(toUTF8(testCase.s1))
-      val hash2 = collation.hashFunction.applyAsLong(toUTF8(testCase.s1))
-      assert(hash1 == hash2)
```
Comment (panbingkun) [inline common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala:158]: Obviously, comparing `testCase.s1` and `testCase.s2` is meaningless.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs).

Nice catch and refactoring, @panbingkun .

**Comment by dongjoon-hyun** [comment]
Also, cc @dbatomic , @cloud-fan , @MaxGekk from the original PR.
- #44968

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**